cli/webhook.py

```diff
@@ -0,0 +1,124 @@
+import sys
+import slackweb
+from sets import Set
+from datetime import datetime
+from cli.settings import Settings
+from cli.appconfig import AppConfig
+
+
+class WebHook:
+
+    flag = False
+
+    def __init__(self):
+        '''this flag makes sure if initialization fails, many hook
+        steps wont try to post message again and again '''
+        self.disabled = True
+        self.emoji = ':rocket:'
+        self.settingObj = Settings()
+        self.appconfigObj = AppConfig()
+
+    def webhookSetting(self):
+
+        config_dir = self.settingObj.getConfigDir()
+        roger_env = self.appconfigObj.getRogerEnv(config_dir)
+        try:
+            if 'webhook_url' in roger_env.keys():
+                self.webhookURL = roger_env['webhook_url']
+            if 'default_channel' in roger_env.keys():
+                self.defChannel = roger_env['default_channel']
+            if 'default_username' in roger_env.keys():
+                self.username = roger_env['default_username']
+            if len(self.username) == 0 or len(self.webhookURL) == 0 or len(self.defChannel) == 0:
+                return
+        except (Exception) as e:
+            print("Warning: slackweb basic initialization failed (error: %s).\
+            Not using slack." % e)
+            return
+        try:
+            self.client = slackweb.Slack(url=self.webhookURL)
+        except (Exception) as e:
+            print("Warning: slackweb basic initialization failed (error: %s).\
+            Not using slack." % e)
+            return  # disabled flag remains False
+        self.disabled = False
```

Comment:
Same as above , any specific reason for self.disabled of not being in try block ?